tuning/utils/preprocessing_utils.py

```diff
@@ -33,8 +34,53 @@
 JSON_OUTPUT_KEY = "output"
 
 
+# check if the provided dataset is pretokenized or not
+# the check is taken from trl
+# https://github.com/huggingface/trl/blob/ddf4c8dc3ecf6d9ee2b24f94c62182ffd682c808/trl/trainer/sft_trainer.py#L498-L509
+def is_pretokenized_dataset(data: Union[str, Dataset, IterableDataset]):
+    if not data:
+        return False
+    if isinstance(data, str):
+        try:
+            data = datasets.load_dataset("json", data_files=data, split="train[:1]")
+        except DatasetGenerationError as e:
+            raise DatasetGenerationError("failed to load the provided dataset") from e
+
+    return ("input_ids" in data.column_names) and ("labels" in data.column_names)
+
+
 def validate_data_args(data_args: configs.DataArguments, packing: bool):
 
+    is_train_data_pretokenized = is_pretokenized_dataset(data_args.training_data_path)
+    is_eval_data_pretokenized = is_pretokenized_dataset(data_args.validation_data_path)
+
+    # if the provided train dataset is pretokenized
+    # however user provides formatting flags, error out
+    if is_train_data_pretokenized:
+        if (
+            data_args.response_template
+            or data_args.data_formatter_template
+            or data_args.dataset_text_field
+        ):
+            logger.warning(
```

Comment:
should be a valueerror . If response template is provided, we will return completionLMcollator. There is no need to allow a response template, it is an incorrect argument.
I am committing this change

Reply:
sure @Ssukriti thanks